version.h:25:37: error: 'c283e79' undeclared (first use in this function)

I just hit this problem in one of my PR's that @grondo hit in #2014.   All the ubuntu, but not centos, travis builders failed with
```
  CC       libflux_la-version.lo
In file included from version.c:15:0:
version.c: In function 'flux_core_version':
version.h:25:37: error: 'c283e79' undeclared (first use in this function)
 #define FLUX_CORE_VERSION_MAJOR     c283e79
                                     ^
version.c:25:18: note: in expansion of macro 'FLUX_CORE_VERSION_MAJOR'
         *major = FLUX_CORE_VERSION_MAJOR;
                  ^~~~~~~~~~~~~~~~~~~~~~~
version.h:25:37: note: each undeclared identifier is reported only once for each function it appears in
 #define FLUX_CORE_VERSION_MAJOR     c283e79
                                     ^
version.c:25:18: note: in expansion of macro 'FLUX_CORE_VERSION_MAJOR'
         *major = FLUX_CORE_VERSION_MAJOR;
                  ^~~~~~~~~~~~~~~~~~~~~~~
version.c:29:41: error: expected expression before ';' token
         *patch = FLUX_CORE_VERSION_PATCH;
                                         ^
In file included from version.c:15:0:
version.h:33:63: error: expected expression before '<<' token
                                      (FLUX_CORE_VERSION_PATCH << 0))
                                                               ^
version.c:30:12: note: in expansion of macro 'FLUX_CORE_VERSION_HEX'
     return FLUX_CORE_VERSION_HEX;
            ^~~~~~~~~~~~~~~~~~~~~
version.c:31:1: error: control reaches end of non-void function [-Werror=return-type]
 }
 ^
```

Configure output this:
```
checking Major version... c283e79
checking Minor version... 9
checking Point version...

--- a/.travis.yml
+++ b/.travis.yml
@@ -122,6 +122,9 @@ before_install:
   fi
 
 script:
+ # Unshallow repository so git describe works.
+ # (The one inside docker-run-checks may fail if git version is too old)
+ - git fetch --unshallow --tags
  - |
   src/test/docker/docker-run-checks.sh -j2 \
     --image=${IMG} \

